### DISCUSSION TITLE
is regex available for ipxe scripting?

### DISCUSSION THREAD

**Discussion by korekhov** [ipxe/ipxe#1238] [General]
Hello there!

I've tried this on the latest version of iPXE, but maybe I'm missing something - is there a way for me to implement something like this?

```
#!ipxe
dhcp
# try httpboot first
iseq ${filename} ^http.* && chain ${filename} ||
# fallback to tftp
chain tftp://${next-server}/pxelinux.cfg/default.ipxe ||
```
Could not find anything like that in the documentation and several simple attempts with specifying a regex pattern did not work...
In some old discussions people were asking about regex as well, it just seems that it was never implemented.

Maybe some other way to implement the above logic?

Thanks!

**Comment by mcb30** [comment]
No, there is no support for regular expression in iPXE, and it is not likely ever to be added.

The logic that you are scripting there is pretty close to what the default behaviour will be anyway. Either the default boot flow (with no embedded script) or the [`autoboot`](https://ipxe.org/cmd/autoboot) command will cause iPXE to attempt booting close to the way you describe: if `${filename}` is an absolute URI (complete with a scheme such as `http://` or `https://`) then attempt a boot from that URI, otherwise attempt to construct a URI from `${next-server}`.

**Comment by korekhov** [comment]
> "No, there is no support for regular expression in iPXE, and it is not likely ever to be added."

Oh, that's too bad, but I'm sure there's a good reason for that - I'm sure there are lots of use-cases could be benefiting from it...

I think one key piece of information I left out - that little script is my (simplified) embedded script that I include into my own build of iPXE along with company CA and client certs.

So, what ends up happening is that your method with "autoboot" and mine (with dhcp+chain ${filename}) both work perfectly fine when ${filename} is set to some URL (and both http:// and https:// work just great). But when ${filename} is NOT an URL, this setup gets into a loop.

Basically, I was trying to extend and add a TFTP-based fallback option to scenario described here - https://ipxe.org/howto/dhcpd#pxe_chainloading

Anyway, I probably just trying to do too much from a single embedded and need to separate that into a couple of scripts - one still embedded and the other that would be loaded and have a little bit more logic.

Thanks!
